python/strings_udf/strings_udf/__init__.py

```diff
@@ -41,59 +28,67 @@ def _get_appropriate_file(sms, cc):
         return None
 
 
+def maybe_patch_numba_linker(driver_version):
+    # Numba thinks cubinlinker is only needed if the driver is older than the ctk
+    # but when strings_udf is present, it might also need to patch because the PTX
+    # file strings_udf relies on may be newer than the driver as well
+    if driver_version < STRINGS_UDF_PTX_VERSION:
+        logger.debug(
+            "Driver version %s.%s needs patching due to strings_udf"
+            % driver_version
+        )
+        if _numba_version_ok:
+            logger.debug("Patching Numba Linker")
+            Linker.new = new_patched_linker
+        else:
+            logger.debug("Cannot patch Numba Linker - unsupported version")
+
+
+def get_ptx_file():
+    dev = cuda.get_current_device()
```

Comment:
We can't make this call (initializing a context) if `RAPIDS_NO_INITIALIZE` is set (see https://nvidia.slack.com/archives/CGLPWR4HJ/p1586206152076500 for a previous similar discussion).